### ISSUE TITLE
DimensionMismatch("array could not be broadcast to match destination")

### ISSUE THREAD

**Issue by JiangXL** [tlnagy/OMETIFF.jl#60]
Load Micromanager .ome.tif1x8x2560x2160x1(CTXYZ), but fail to match dimension. Do wrong `.ome.tif` tag trigger this error. More, Julia can load image to 2160x2560x8 array when I change `.ome.tif` to `.tif` to by pass OMETIFF

FYI

```julia
img_test = load("/RawData/20200102/leaking chip/MMStack_Pos0.ome.tif")
┌ Warning: ignored the empty prefix for 'http://www.openmicroscopy.org/Schemas/OME/2015-01'; expected to be non-empty
└ @ EzXML /home/hf/.julia/packages/EzXML/QtGgF/src/xpath.jl:85
Error encountered while loading "/RawData/20200102/leaking chip/MMStack_Pos0.ome.tif".

Fatal error:
DimensionMismatch("array could not be broadcast to match destination")

Stacktrace:
 [1] handle_error(::DimensionMismatch, ::File{DataFormat{:OMETIFF}}) at /home/hf/.julia/packages/FileIO/I1ONY/src/error_handling.jl:82
 [2] handle_exceptions(::Array{Any,1}, ::String) at /home/hf/.julia/packages/FileIO/I1ONY/src/error_handling.jl:77
 [3] #load#27(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(load), ::FileIO.Formatted) at /home/hf/.julia/packages/FileIO/I1ONY/src/loadsave.jl:189
 [4] load(::FileIO.Formatted) at /home/hf/.julia/packages/FileIO/I1ONY/src/loadsave.jl:169
 [5] #load#13(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(load), ::String) at /home/hf/.julia/packages/FileIO/I1ONY/src/loadsave.jl:118
 [6] load(::String) at /home/hf/.julia/packages/FileIO/I1ONY/src/loadsave.jl:118
 [7] top-level scope at In[5]:1
```

```julia
println(OMETIFF.dump_omexml("/RawData/20200102/leaking chip/MMStack_Pos0.ome.tif"))
<OME xmlns="http://www.openmicroscopy.org/Schemas/OME/2015-01" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.openmicroscopy.org/Schemas/OME/2015-01 http://www.openmicroscopy.org/Schemas/OME/2015-01/ome.xsd">
  <Instrument ID="Instrument:0">
    <Detector ID="Detector:Andor sCMOS Camera" Model="Zyla 5.5 CL 3 Tap" SerialNumber="VSC-00788"/>
  </Instrument>
  <Image ID="Image:0" Name="MMStack_Pos0">
    <InstrumentRef ID="Instrument:0"/>
    <StageLabel Name="pos0" X="0.0" XUnit="µm" Y="0.0" YUnit="µm"/>
    <Pixels BigEndian="false" DimensionOrder="XYCZT" ID="Pixels:0" Interleaved="true" SizeC="1" SizeT="8" SizeX="2560" SizeY="2160" SizeZ="1" Type="uint16">
      <Channel ID="Channel:0:0" Name="" SamplesPerPixel="1">
        <LightPath/>
      </Channel>
      <TiffData FirstC="0" FirstT="0" FirstZ="0" IFD="0" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="1" FirstZ="0" IFD="1" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="2" FirstZ="0" IFD="2" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="3" FirstZ="0" IFD="3" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="4" FirstZ="0" IFD="4" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="5" FirstZ="0" IFD="5" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="6" FirstZ="0" IFD="6" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <TiffData FirstC="0" FirstT="7" FirstZ="0" IFD="7" PlaneCount="1">
        <UUID FileName="MMStack_Pos0.ome.tif">urn:uuid:5a8594cd-4c87-4319-accb-79c766dd7b21</UUID>
      </TiffData>
      <Plane DeltaT="6.3129650892522E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="0" TheZ="0"/>
      <Plane DeltaT="6.31296604946E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="1" TheZ="0"/>
      <Plane DeltaT="6.312966129446E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="2" TheZ="0"/>
      <Plane DeltaT="6.3129661469683E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="3" TheZ="0"/>
      <Plane DeltaT="6.3129667945224E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="4" TheZ="0"/>
      <Plane DeltaT="6.312968457888E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="5" TheZ="0"/>
      <Plane DeltaT="6.3129684689447E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="6" TheZ="0"/>
      <Plane DeltaT="6.3129684846153E11" DeltaTUnit="ms" PositionX="0.0" PositionXUnit="µm" PositionY="0.0" PositionYUnit="µm" PositionZ="0.0" PositionZUnit="µm" TheC="0" TheT="7" TheZ="0"/>
    </Pixels>
  </Image>
</OME>
```

**Comment by tlnagy** [comment]
Hmm. This could potentially be a bug. There doesn't seem to be anything weird with your OME-XML, pretty standard stuff.

What version of OMETIFF and Julia are you using (could you give the output of `] status` in the REPL)?

Additionally could you run the following command in the REPL:

```julia
julia> using OMETIFF, FileIO

julia> filepath = "path to your file .... "

julia> load(Stream(format"OMETIFF", open(filepath), filepath))
```

this should give us the full stacktrace instead of the FileIO error that you got that isn't very helpful. Alternatively, if you prefer, you can send me the image via Dropbox, Google Drive, etc and I can diagnose on my end.

**Comment by JiangXL** [comment]
```Julia
(v1.2) pkg> status
    Status `~/.julia/environments/v1.2/Project.toml`
  [336ed68f] CSV v0.5.21
  [c5f51814] CUDAdrv v5.0.1
  [be33ccc6] CUDAnative v2.7.0
  [3a865a2d] CuArrays v1.6.0
  [41ba435c] Deconvolution v0.3.0
  [5789e2e9] FileIO v1.2.1
  [f67ccb44] HDF5 v0.12.5
  [7073ff75] IJulia v1.20.2
  [4381153b] ImageDraw v0.2.2
  [92ff4b2b] ImageFeatures v0.3.0
  [6218d12a] ImageMagick v0.7.5
  [86fae568] ImageView v0.10.3
  [916415d5] Images v0.19.3
  [9b13fd28] IndirectArrays v0.5.1
  [15e1cf62] NPZ v0.4.0
  [2d0ec36b] OMETIFF v0.3.4
  [e0fc9d43] PkgMirrors v1.3.0
  [91a5bcdd] Plots v0.28.4
  [438e738f] PyCall v1.91.2
  [d330b81b] PyPlot v2.8.2
  [77f5812f] SingularSpectrumAnalysis v0.2.1
  [2913bbd2] StatsBase v0.32.0
  [4abe2ffd] ThreeDeconv v0.0.0 #master (https://github.com/computational-imaging/ThreeDeconv.jl.git)
  [de0858da] Printf 
```

**Comment by JiangXL** [comment]
Unluckily, it still be same stacktrace 
```julia
julia> using OMETIFF, FileIO
[ Info: Recompiling stale cache file /home/hf/.julia/compiled/v1.2/OMETIFF/Bj5Qs.ji for OMETIFF [2d0ec36b-e807-5756-994b-45af29551fcf]

julia> img = "/igem/RawData/20200102/leaking chip/MMStack_Pos0.ome.tif"
"/igem/RawData/20200102/leaking chip/MMStack_Pos0.ome.tif"

julia> load(Stream(format"OMETIFF", open(img), img))
┌ Warning: ignored the empty prefix for 'http://www.openmicroscopy.org/Schemas/OME/2015-01'; expected to be non-empty
└ @ EzXML ~/.julia/packages/EzXML/QtGgF/src/xpath.jl:85
Error encountered while loading "/igem/RawData/20200102/leaking chip/MMStack_Pos0.ome.tif".

Fatal error:
ERROR: DimensionMismatch("array could not be broadcast to match destination")
Stacktrace:
 [1] handle_error(::DimensionMismatch, ::Stream{DataFormat{:OMETIFF},IOStream}) at /home/hf/.julia/packages/FileIO/I1ONY/src/error_handling.jl:82
 [2] handle_exceptions(::Array{Any,1}, ::String) at /home/hf/.julia/packages/FileIO/I1ONY/src/error_handling.jl:77
 [3] #load#27(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(load), ::Formatted) at /home/hf/.julia/packages/FileIO/I1ONY/src/loadsave.jl:189
 [4] load(::Formatted) at /home/hf/.julia/packages/FileIO/I1ONY/src/loadsave.jl:169
 [5] top-level scope at REPL[4]:1
```

**Comment by JiangXL** [comment]
Thank for your kind. You can find this image in [Google Drive](https://drive.google.com/file/d/1S0K1dRrlkPr65msT4hMgGf1UZtbZ8K4h/view?usp=sharing)

**Comment by tlnagy** [comment]
That's what I get for not actually testing my own command, the last line should've been:

```julia
OMETIFF.load(Stream(format"OMETIFF", open(img), img))
```

I think I tracked down the bug and fixed it in #61. If you're okay with it, I would like to add your image to the test suite so that any regressions may be caught in the future. What do you say?

**Comment by JiangXL** [comment]
I'm happy to improve OMETIFF.jl

**Comment by tlnagy** [comment]
I've tagged a new version for release https://github.com/JuliaRegistries/General/pull/7595 (v0.3.5) with the fixes from #61. Once that merges, running `] update` in the REPL should pull the new version.
